In [4]:
# Standard data science libraries
import pandas as pd
import numpy as np
from pathlib import Path
from sklearn.preprocessing import MultiLabelBinarizer

import warnings
warnings.filterwarnings('ignore')

In [139]:
from bnb.data_processing.read import read_csvfile
from bnb.data_processing.clean import price_format
from bnb.data_processing.clean import bathrooms_int
from bnb.data_processing.clean import shared_bathroom
from bnb.data_processing.clean import super_host
from bnb.data_processing.clean import beds_int
from bnb.data_processing.clean import bedrooms_int
from bnb.data_processing.clean import bedrooms_int_final
from bnb.data_processing.clean import num_of_stars
from bnb.data_processing.clean import entire_shared_private_room
from bnb.data_processing.clean import bed_type
from bnb.data_processing.clean import amenities_to_vars
from bnb.data_processing.clean import encode_amenities
from bnb.data_processing.clean import one_hot_encode_bedtype

## Reading Data

In [6]:
Data_dir = Path('C:/Users/KBrig/AirBnBRepo/AirBnB/data')
Data_dir

WindowsPath('C:/Users/KBrig/AirBnBRepo/AirBnB/data')

In [54]:
NewYork_CSV = Data_dir/'AirBnBNY.csv'
Bangkok_CSV = Data_dir/'AirBnbDataBangkok.csv'
Cairo_CSV = Data_dir/'AirBnbDataCairo.csv'


## NewYork

In [8]:
NewYorkData = read_csvfile(NewYork_CSV)
NewYorkData.head()


Unnamed: 0       Accuracy_stars  \
0           0    Rating 5 out of 5   
1           1  Rating 4.5 out of 5   
2           2    Rating 5 out of 5   
3           3    Rating 5 out of 5   
4           4    Rating 5 out of 5   

                                           Amenities     Bathrooms_num  \
0  ['Indoor fireplace', 'Wifi', 'Iron', 'Laptop f...            1 bath   
1  ['Indoor fireplace', 'Wifi', 'Laptop friendly ...     1 shared bath   
2  ['Wifi', 'Essentials', 'Heating', 'Air conditi...     1 shared bath   
3  ['Wifi', 'Iron', 'Laptop friendly workspace', ...    2 shared baths   
4  ['Wifi', 'Washer', 'Dryer', 'Iron', 'Laptop fr...  1.5 shared baths   

  Bedrooms_no Beds_no              CheckIn        CheckIn_stars  \
0   1 bedroom   1 bed  2019-03-21 00:00:00    Rating 5 out of 5   
1   1 bedroom   1 bed  2019-03-21 00:00:00  Rating 4.5 out of 5   
2   1 bedroom   1 bed  2019-03-21 00:00:00    Rating 5 out of 5   
3   1 bedroom   1 bed  2019-03-21 00:00:00    Rating 5 out of 5   
4   1 bedroom   1 bed  2019-03-21 00:00:00    Rating 5 out of 5   

              CheckOut  Cleanliness_stars         ...           Latitude  \
0  2019-03-26 00:00:00  Rating 5 out of 5         ...           40.73321   
1  2019-03-26 00:00:00  Rating 4 out of 5         ...           40.76580   
2  2019-03-26 00:00:00  Rating 5 out of 5         ...           40.75972   
3  2019-03-26 00:00:00  Rating 5 out of 5         ...           40.81386   
4  2019-03-26 00:00:00  Rating 5 out of 5         ...           40.69284   

        Location_stars  Longitude  Price  \
0    Rating 5 out of 5  -74.00242   $244   
1    Rating 5 out of 5  -73.95812    $62   
2  Rating 4.5 out of 5  -74.03827    $44   
3    Rating 5 out of 5  -73.95936    $78   
4  Rating 4.5 out of 5  -73.94418    $80   

                                              Review  Reviewers_no  \
0  87 Reviews\r\nAccuracy\r\nCommunication\r\nCle...           NaN   
1  5 Reviews\r\nAccuracy\r\nCommunication\r\nClea...           NaN   
2  83 Reviews\r\nAccuracy\r\nCommunication\r\nCle...           NaN   
3  32 Reviews\r\nAccuracy\r\nCommunication\r\nCle...           NaN   
4  28 Reviews\r\nAccuracy\r\nCommunication\r\nCle...           NaN   

          Sleeping_engagment              Super_host  \
0   Bedroom 1\r\n1 queen bed  Brian is a Superhost ·   
1  Bedroom 1\r\n1 single bed    Samet is a Superhost   
2   Bedroom 1\r\n1 queen bed  Rebecca is a Superhost   
3   Bedroom 1\r\n1 queen bed    James is a Superhost   
4                        NaN      Cat is a Superhost   

                                               Title          Value_stars  
0  *Luxury 1 Bedrm Private Designer Home & Roofdeck*  Rating 4.5 out of 5  
1        Single Bedroom-Upper East Side-Central Park    Rating 4 out of 5  
2  Quiet room for sleeping - 15 min from Times Sq...    Rating 5 out of 5  
3        Cozy room C in upper west side NY, Columbia    Rating 5 out of 5  
4                                    Cozy in Bedstuy    Rating 5 out of 5  

[5 rows x 23 columns]

## Bangkok

In [55]:
BangkokData = read_csvfile(Bangkok_CSV)
BangkokData.head()

CheckIn         CheckOut  Has_Weekend  \
0  2019-03-12 0:00  2019-03-17 0:00         True   
1  2019-03-12 0:00  2019-03-17 0:00         True   
2  2019-03-12 0:00  2019-03-17 0:00         True   
3  2019-03-12 0:00  2019-03-17 0:00         True   
4  2019-03-12 0:00  2019-03-17 0:00         True   

                                               Title Price  \
0                Cosy room with 3 mins walk from MRT   $34   
1                                       曼谷市区BTS旁品质公寓   $48   
2  Modern apartment close to city near sky train(...   $27   
3  Outstanding Location in Heart of Bangkok 2mins...   $35   
4  5 Min walk MRT Phraram9, 1stop RodFai Train Ma...   $34   

                 Super_host  Latitude  Longitude    EntireHomeVsRoom  \
0  Thongjuta is a Superhost  13.78165  100.57453    ENTIRE APARTMENT   
1                       NaN  13.69750  100.60708    ENTIRE APARTMENT   
2       Kiek is a Superhost  13.71252  100.46783  ENTIRE CONDOMINIUM   
3                       NaN  13.72032  100.49623  ENTIRE CONDOMINIUM   
4      Patty is a Superhost  13.75738  100.56318    ENTIRE APARTMENT   

  Bedrooms_no                        ...                          \
0   1 bedroom                        ...                           
1   1 bedroom                        ...                           
2   1 bedroom                        ...                           
3   1 bedroom                        ...                           
4         NaN                        ...                           

                              Sleeping_engagment  \
0                         Bedroom 1\n1 queen bed   
1                        Bedroom 1\n1 double bed   
2                         Bedroom 1\n1 queen bed   
3                          Bedroom 1\n1 king bed   
4  Common spaces\n2 queen beds, 1 floor mattress   

                                           Amenities     Accuracy_stars  \
0  ['Wifi', 'Iron', 'Cable TV', 'Laptop friendly ...  Rating 5 out of 5   
1  ['Wifi', 'Indoor fireplace', 'Heating', 'Iron'...                NaN   
2  ['Wifi', 'Heating', 'Laptop friendly workspace...  Rating 5 out of 5   
3  ['Wifi', 'Iron', 'Cable TV', 'Laptop friendly ...  Rating 5 out of 5   
4  ['Wifi', 'Iron', 'Laptop friendly workspace', ...                NaN   

  Communication_stars    Cleanliness_stars     Location_stars  \
0   Rating 5 out of 5  Rating 4.5 out of 5  Rating 5 out of 5   
1                 NaN                  NaN                NaN   
2   Rating 5 out of 5    Rating 5 out of 5  Rating 4 out of 5   
3   Rating 5 out of 5  Rating 4.5 out of 5  Rating 5 out of 5   
4                 NaN                  NaN                NaN   

       CheckIn_stars          Value_stars Reviewers_no  \
0  Rating 5 out of 5    Rating 5 out of 5          NaN   
1                NaN                  NaN          NaN   
2  Rating 5 out of 5  Rating 4.5 out of 5          NaN   
3  Rating 5 out of 5    Rating 5 out of 5          NaN   
4                NaN                  NaN          NaN   

                                              Review  
0  101 Reviews\nAccuracy\nCommunication\nCleanlin...  
1  No reviews (yet)\nBe one of the first guests t...  
2  4 Reviews\nAccuracy\nCommunication\nCleanlines...  
3  3 Reviews\nAccuracy\nCommunication\nCleanlines...  
4  No reviews (yet)\nThis host has 1631 reviews f...  

[5 rows x 22 columns]

## Cairo

In [10]:
CairoData = read_csvfile(Cairo_CSV)
CairoData.head()

CheckIn         CheckOut  Has_Weekend  \
0  2019-03-14 0:00  2019-03-19 0:00         True   
1  2019-03-14 0:00  2019-03-19 0:00         True   
2  2019-03-14 0:00  2019-03-19 0:00         True   
3  2019-03-14 0:00  2019-03-19 0:00         True   
4  2019-03-14 0:00  2019-03-19 0:00         True   

                                               Title Price Super_host  \
0                                         Dad Haus!x   $15        NaN   
1                    Apartment in Rehab City, Cairo.   $12        NaN   
2  Sunny and friendly apartment close to the airport   $22        NaN   
3  Cozy 2 persons budget room in luxurious/safe area   $19        NaN   
4         Girls apartment - Sunny and calm apartment   $23        NaN   

   Latitude  Longitude             EntireHomeVsRoom Bedrooms_no  \
0  29.95961   31.25858    Private room in apartment   1 bedroom   
1  30.05760   31.48779             Entire apartment   1 bedroom   
2  30.06085   31.49540    Private room in apartment   1 bedroom   
3  30.06284   31.51743  Private room in condominium   1 bedroom   
4  30.06263   31.48585    Private room in apartment   1 bedroom   

                         ...                          \
0                        ...                           
1                        ...                           
2                        ...                           
3                        ...                           
4                        ...                           

         Sleeping_engagment  \
0   Bedroom 1\n2 queen beds   
1    Bedroom 1\n1 queen bed   
2                       NaN   
3  Bedroom 1\n2 single beds   
4    Bedroom 1\n1 queen bed   

                                           Amenities     Accuracy_stars  \
0  ['Wifi', 'Indoor fireplace', 'Dryer', 'Laptop ...                NaN   
1  ['Wifi', 'Laptop friendly workspace', 'Washer'...                NaN   
2  ['Wifi', 'Indoor fireplace', 'Cable TV', 'Lapt...  Rating 4 out of 5   
3  ['Wifi', 'Laptop friendly workspace', 'Iron', ...  Rating 5 out of 5   
4  ['Wifi', 'Cable TV', 'Laptop friendly workspac...                NaN   

  Communication_stars    Cleanliness_stars       Location_stars  \
0                 NaN                  NaN                  NaN   
1                 NaN                  NaN                  NaN   
2   Rating 5 out of 5  Rating 3.5 out of 5  Rating 4.5 out of 5   
3   Rating 5 out of 5    Rating 5 out of 5    Rating 5 out of 5   
4                 NaN                  NaN                  NaN   

       CheckIn_stars        Value_stars Reviewers_no  \
0                NaN                NaN          NaN   
1                NaN                NaN          NaN   
2  Rating 5 out of 5  Rating 4 out of 5          NaN   
3  Rating 5 out of 5  Rating 5 out of 5          NaN   
4                NaN                NaN          NaN   

                                              Review  
0  No reviews (yet)\nBe one of the first guests t...  
1  No reviews (yet)\nThis host has 11 reviews for...  
2  6 Reviews\nAccuracy\nCommunication\nCleanlines...  
3  4 Reviews\nAccuracy\nCommunication\nCleanlines...  
4  2 Reviews\nAmelia\nDecember 2018\nThank you ve...  

[5 rows x 22 columns]

## Data Preprocessing

## NewYork

In [11]:
NewYorkData.shape

(3957, 23)

In [12]:
NewYorkData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3957 entries, 0 to 3956
Data columns (total 23 columns):
Unnamed: 0             3957 non-null int64
Accuracy_stars         3614 non-null object
Amenities              3957 non-null object
Bathrooms_num          3922 non-null object
Bedrooms_no            3348 non-null object
Beds_no                3895 non-null object
CheckIn                3957 non-null object
CheckIn_stars          3614 non-null object
CheckOut               3957 non-null object
Cleanliness_stars      3614 non-null object
Communication_stars    3614 non-null object
EntireHomeVsRoom       3957 non-null object
Has_Weekend            3957 non-null bool
Latitude               3957 non-null float64
Location_stars         3614 non-null object
Longitude              3957 non-null float64
Price                  3917 non-null object
Review                 3956 non-null object
Reviewers_no           0 non-null float64
Sleeping_engagment     3158 non-null object
Super_host      

We have a total of 23 features as shown above: 1 boolean (Has_Weekend), 3 numeric, float64(Latitude, Longitude, Reviewers_num), and the rest are saved as data objects. Let's go through them!! 



So, we need to keep ONLY features that can add information to our analysis. Predictors with many null values can be considered redundant and ignore tham.

### Missing Data

In [13]:
NewYorkData.isnull().sum()

Unnamed: 0                0
Accuracy_stars          343
Amenities                 0
Bathrooms_num            35
Bedrooms_no             609
Beds_no                  62
CheckIn                   0
CheckIn_stars           343
CheckOut                  0
Cleanliness_stars       343
Communication_stars     343
EntireHomeVsRoom          0
Has_Weekend               0
Latitude                  0
Location_stars          343
Longitude                 0
Price                    40
Review                    1
Reviewers_no           3957
Sleeping_engagment      799
Super_host             2010
Title                     0
Value_stars             343
dtype: int64

In [14]:
## Features with large number of missing data
[column for column in NewYorkData.columns if NewYorkData[column].count() / len(NewYorkData) <= 0.7]

['Reviewers_no', 'Super_host']

So "Super_host" can have two possible values: Yes/True or No/False. Samples with null values are actually representing non super_host. On the other hand, having 3957 of null values for the Reviews_no says that there is something wrong with collecting data for this variable. So, it is better to exclude it.

Also, for the reviews' features, we see that we have the same number of null values for the Accuracy,CheckIn,Cleanliness,Communication, Location and Value Stars. This make sense since this number points to the homes with "no reviews yet".  We are going to have this invistigation for all features!

Let's classify the features: numeric, catogerical,etc.


So we have 23 variables six of them are categorical variables reflecting the clients' satisfaction in different aspects: 'Accuracy_stars','CheckIn_stars','Cleanliness_stars','Communication_stars','Location_stars','Value_stars'.

Other categorical variables such as 'EntireHomeVsRoom','Has_Weekend', and 'Sleeping_engagment'.

We also have dates: 'CheckIn','CheckOut'.

We have three numeric features: 'Price','Bathrooms_num', 'Bedrooms_no', 'Beds_no', 'Latitude','Longitude'.

rest:'Amenities', 'Review', 'Title', and 'Super_host'


Let's tidy up!


In [15]:
NewYorkData.drop(['Unnamed: 0','Reviewers_no'],axis=1, inplace=True)

### Does it have a weekend?

We have originally one boolean variable "Has_Weekend". Let's check its levels!

In [16]:
NewYorkData['Has_Weekend'].unique()

array([ True])

So "Has_Weekend" doesn't say more than every time_period is a mix of weekdays and weekends. Thus, we can ignore it!

In [17]:
NewYorkData.drop('Has_Weekend',axis=1, inplace=True)

### Is Super Host?

In [18]:
NewYorkData.Super_host[0:10]

0    Brian is a Superhost ·
1      Samet is a Superhost
2    Rebecca is a Superhost
3      James is a Superhost
4        Cat is a Superhost
5       Rick is a Superhost
6                       NaN
7    Shailah is a Superhost
8                       NaN
9       Obed is a Superhost
Name: Super_host, dtype: object

In [19]:
NewYorkData['Super_host'] = super_host(NewYorkData)

True = 1, False = 0

In [20]:
NewYorkData['Super_host'][0:10]

0    1
1    1
2    1
3    1
4    1
5    1
6    0
7    1
8    0
9    1
Name: Super_host, dtype: int64

## Numeric Data

### Price

In [21]:
print(type(NewYorkData['Price'][0]))
print(NewYorkData['Price'][0:10])

<class 'str'>
0    $244
1     $62
2     $44
3     $78
4     $80
5    $116
6     $96
7    $119
8     $73
9     $63
Name: Price, dtype: object


In [22]:
NewYorkData['Price'] = price_format(NewYorkData)

In [23]:
print(type(NewYorkData['Price'][0]))
print(NewYorkData['Price'][0:10])

<class 'numpy.float64'>
0    244.0
1     62.0
2     44.0
3     78.0
4     80.0
5    116.0
6     96.0
7    119.0
8     73.0
9     63.0
Name: Price, dtype: float64


### Number of Bathrooms

In [24]:
print(type(NewYorkData['Bathrooms_num'][0]))
print(NewYorkData['Bathrooms_num'].unique())

<class 'str'>
['1 bath' '1 shared bath' '2 shared baths' '1.5 shared baths'
 '1.5 private baths' '1 private bath' '2 private baths' nan
 '2.5 shared baths' '2 baths' '2.5 private baths' '3 shared baths'
 '4 shared baths' '0 baths' 'Half-bath' 'Shared half-bath' '1.5 baths'
 '2.5 baths' '3 baths' '0 shared baths' '3.5 shared baths']


In [25]:
NewYorkData['Bathrooms_num_int']= float(0)

In [26]:
NewYorkData['Bathrooms_num_int'] = bathrooms_int(NewYorkData)

In [27]:
for i in range(10):
    print(NewYorkData['Bathrooms_num'][i],NewYorkData['Bathrooms_num_int'][i])

1 bath 1.0
1 shared bath 1.0
1 shared bath 1.0
2 shared baths 2.0
1.5 shared baths 1.5
1 shared bath 1.0
1 shared bath 1.0
1 shared bath 1.0
1 shared bath 1.0
2 shared baths 2.0


### Shared vs. Private Bathroom

In [28]:
NewYorkData['shared_bathroom']= shared_bathroom(NewYorkData)

In [29]:
NewYorkData[['Bathrooms_num','shared_bathroom']][0:10]

Bathrooms_num  shared_bathroom
0            1 bath            False
1     1 shared bath             True
2     1 shared bath             True
3    2 shared baths             True
4  1.5 shared baths             True
5     1 shared bath             True
6     1 shared bath             True
7     1 shared bath             True
8     1 shared bath             True
9    2 shared baths             True

### Number of Bedrooms

In [30]:
print(type(NewYorkData['Bedrooms_no'][0]))
print(NewYorkData['Bedrooms_no'].unique())

<class 'str'>
['1 bedroom' 'Super cute 1 bedroom apartment close to TimeSquare' nan
 'Large private bedroom 1 stop from Manhattan'
 '1 large bedroom HARLEM/NYC/terrace/'
 'Cozy tiny bedroom w/large living-room & kitchen.' '2 bedrooms'
 'Beautiful bedroom close to NYC' '3 bedrooms'
 'Spacious 3 bedroom apartment in WILLIAMSBURG'
 'Urban Farmers bedroom complete with fort' '4 bedrooms'
 'Cozy bedroom in townhouse with backyard in LIC'
 'Convenient Private bedroom 10 MIN FROM NYC'
 'Cozy bedroom with balcony. 25min TimeSquare'
 'Sunny, Clean, Nice bedroom available in Bushwick'
 'Large bedroom/Heart of Brooklyn'
 'Private bedroom in the Upper West Side, Manhattan'
 'Large queen size bedroom in huge loft apt'
 'Nice bedroom in east village apartment'
 'Very large and comfortable Manhattan one bedroom.'
 'Oasis bedroom for solo travelers' 'Big bedroom with queen size Bed'
 'Cozy bedroom by Yankee Stadium'
 'Private bedroom&bathroom in the heart of Manhattan'
 'Huge private bedroom for quiet

In [31]:
NewYorkData['Bedrooms_int'] = None

In [33]:
NewYorkData['Bedrooms_int']= bedrooms_int(NewYorkData)

In [34]:
for i in range(len(NewYorkData)):
    print(NewYorkData['Bedrooms_no'][i],NewYorkData['Bedrooms_int'][i])

1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
Super cute 1 bedroom apartment close to TimeSquare 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
Large private bedroom 1 stop from Manhattan private
1 large bedroom HARLEM/NYC/terrace/ large
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
Cozy tiny bedroom w/large living-room & kitchen. tiny
1 bedroom 1
1 bedroom 1
2 bedrooms 2
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
Beautiful bedroom close to NYC Beautiful
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedro

1 bedroom 1
1 bedroom 1
Private bedroom in a loft apartment (Williamsburg) Private
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom with 3 beds in the heart of Times Square 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
4 bedrooms 4
1 bedroom 1
Lovely 2-bedroom 1bath in Chinatown & Little Italy Lovely
nan -1
1 bedroom 1
Sunny with 2 large bedrooms large
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
2 bedrooms 2
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
2 bedrooms 2
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
Amazing 3 bedroom in W

nan -1
1 bedroom 1
Prospect Park 3 bedroom, Sleeps 8 3
1 bedroom 1
1 bedroom 1
3 bedrooms 3
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
Times Square/ Hell's Kitchen HUGE 1 bedroom Apt 1
nan -1
nan -1
nan -1
3 bedrooms 3
1 bedroom 1
1 bedroom 1
The studio is located on the 6th floor which is the top floor of an old industrial building in downtown New York City. With rare high 11.5 ft. ceilings, bright western exposure, oversized window with expansive views. You can watch the clouds roll in from the west and view the construction of the new Freedom Tower. A rare view in NYC. located in Manhattan’s lower east side the cultural hubub of the city. Chinatown offers a variety of authentic cuisine, out door markets and fun specialty shops. If your an early riser in the morning the locals practice tai chi in the nearby park; wether your interested in participating or just a spectator. Historically, the oldest synagogue is right down the block along with

1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
2 bedrooms 2
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
Comfortable and reasonable price bedroom price
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
Sunny, private bedroom, bathroom in Red Hook Bklyn private
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
nan -1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
One bedroom Midtown East Quiet Sanctuary One
1 bedroom 1
2 bedrooms 2
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
2 bedrooms 2
1 bedroom 1
1 bedroom 1
1 bedroom 1
Bright bedroom close to NYC- Breakfast included! 2 Bright
nan -1
1 bedroom 1
1 bedroom 1
nan -1
1 bed

nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
Times Square/ Hell's Kitchen HUGE 1 bedroom Apt 1
2 bedrooms 2
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
2 bedrooms 2
3 bedrooms 3
1 bedroom 1
nan -1
2 bedrooms 2
1 bedroom 1
1 bedroom 1
1 bedroom 1
2 bedrooms 2
3 bedrooms 3
nan -1
nan -1
1 bedroom 1
3 bedrooms 3
1 bedroom 1
2 bedrooms 2
2 bedrooms 2
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
2 bedrooms 2
2 bedrooms 2
3 bedrooms 3
Bright 1 bedroom apartment in the Village 1
1 bedroom 1
3 bedrooms 3
Modern, Park Slope, entire one-bedroom apartment entire
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
Best of SOHO Gorgeous 2 bedroom 2
1 bedroom 1
1 bedroom 1
1 bedroom 1
Convenient Private bedroom 10 MIN FROM NYC Private
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
2 bedrooms 2
Cozy 1 bedroom/ Minutes from NY 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
4 bedrooms 4
nan -1
1 bedroom 1
Step-free access to the bedroom the
1 bed

1 bedroom 1
nan -1
1 bedroom 1
Cozy 1 bedroom/ Minutes from NYC 1
nan -1
nan -1
nan -1
1 bedroom 1
2 bedrooms 2
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
New 1 bedroom with pool 10 min from Manhattan 1
1 bedroom 1
1 bedroom 1
2 bedrooms 2
1 bedroom 1
nan -1
2 bedrooms 2
5 bedrooms 5
2 bedrooms 2
1 bedroom 1
2 bedrooms 2
1 bedroom 1
nan -1
1 bedroom 1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
nan -1
1 bedroom 1
nan -1
nan -1
nan -1
1 bedroom 1
1 bedroom 1
1 bedroom 1
1 bedroom 1
3 bedrooms 3
2 bedrooms 2
1 bedroom 1
nan -1
1 bedroom 1
nan -1


In [35]:
NewYorkData.Bedrooms_int.unique()

array(['1', -1, 'private', 'large', 'tiny', '2', 'Beautiful', '3',
       'Farmers', '4', 'Cozy', 'Private', 'Nice', 'Large', 'size', 'one',
       'Oasis', 'Big', 'big', 'entire', 'One', 'loft', 'master', 'two',
       'Lovely', '5', 'spacious', '"loft"', 'the', 'artistic', 'calm',
       'three', 'Two', 'Midtown', '6', 'Sunny', 'Astoria', 'Comfortable',
       'Peaceful', 'Huge', 'Luxury', 'price', 'Bright', 'our'],
      dtype=object)

In [36]:
NewYorkData[NewYorkData.Bedrooms_int == 'our']['EntireHomeVsRoom']

2668    ROOM IN BOUTIQUE HOTEL
Name: EntireHomeVsRoom, dtype: object

In [37]:
NewYorkData[NewYorkData.Bedrooms_int == 'Astoria']['EntireHomeVsRoom']

2088    PRIVATE ROOM IN APARTMENT
Name: EntireHomeVsRoom, dtype: object

In [38]:
NewYorkData[NewYorkData.Bedrooms_int == 'Oasis']['EntireHomeVsRoom']

331    ENTIRE APARTMENT
Name: EntireHomeVsRoom, dtype: object

In [39]:
oneList = ['private', 'large','tiny','Beautiful', 'Farmers','Cozy', 'Private', 'Nice', 'Large', 'size', 'one',
       'Oasis', 'Big', 'big', 'entire', 'One', 'loft', 'master','Lovely','spacious', 'the', 'artistic', 'calm', 'Midtown','Sunny', 'Astoria', 'Comfortable',
       'Peaceful', 'Huge', 'Luxury', 'price', 'Bright', 'our']
for i in oneList:
    NewYorkData['Bedrooms_int']= NewYorkData['Bedrooms_int'].str.replace(i,'1')
    
twoList =['two','Two']
for i in twoList:
    NewYorkData['Bedrooms_int']= NewYorkData['Bedrooms_int'].str.replace(i,'2')
    

NewYorkData['Bedrooms_int']= NewYorkData['Bedrooms_int'].str.replace('three','3')
NewYorkData['Bedrooms_int']= NewYorkData['Bedrooms_int'].str.replace("loft",'1')

In [40]:
NewYorkData.Bedrooms_int.unique()

array(['1', nan, '2', '3', '4', '5', '"1"', '6'], dtype=object)

In [41]:
NewYorkData['Bedrooms_int']= NewYorkData['Bedrooms_int'].str.replace('"1"','1')

In [42]:
NewYorkData['Bedrooms_int'] = bedrooms_int_final(NewYorkData)
#for i in range(len(NewYorkData)):
#    if(NewYorkData.Bedrooms_int[i]==None):
#        NewYorkData.Bedrooms_int[i]= 0
#    elif(NewYorkData.Bedrooms_int[i] == '"1"'):
#        NewYorkData.Bedrooms_int[i]= 1
#    else:
#        NewYorkData.Bedrooms_int[i]=int(NewYorkData.Bedrooms_int[i])

### Number of Beds

In [45]:
print(type(NewYorkData['Beds_no'][0]))
print(NewYorkData['Beds_no'].unique())

<class 'str'>
['1 bed' '0 beds' 'Obed' '2 beds' 'PRIVATE ROOM IN BED AND BREAKFAST'
 '3 beds' nan '5 beds' '4 beds' '8 beds'
 'SHARED ROOM IN BED AND BREAKFAST' '6 beds' 'ENTIRE BED AND BREAKFAST']


In [46]:
NewYorkData['Beds_int'] = float(0)

In [47]:
NewYorkData['Beds_int'] = beds_int(NewYorkData)

In [48]:
for i in range(10):
    print(NewYorkData['Beds_no'][i],NewYorkData['Beds_int'][i])

1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
0 beds 0.0
1 bed 1.0
Obed 0.0


### Latitude

In [49]:
print(type(NewYorkData['Latitude'][0]))
print(NewYorkData['Latitude'].unique())

<class 'numpy.float64'>
[40.73321 40.7658  40.75972 ... 40.77264 40.76406 40.73354]


### Longitude

In [50]:
print(type(NewYorkData['Longitude'][0]))
print(NewYorkData['Longitude'].unique())

<class 'numpy.float64'>
[-74.00242 -73.95812 -74.03827 ... -74.03762 -74.02755 -73.90769]


Thanks god! we have Latitude and Longitude ready to go!

### Bangkok

In [56]:
BangkokData.columns

Index(['CheckIn', 'CheckOut', 'Has_Weekend', 'Title', 'Price', 'Super_host',
       'Latitude', 'Longitude', 'EntireHomeVsRoom', 'Bedrooms_no', 'Beds_no',
       'Bathrooms_num', 'Sleeping_engagment', 'Amenities', 'Accuracy_stars',
       'Communication_stars', 'Cleanliness_stars', 'Location_stars',
       'CheckIn_stars', 'Value_stars', 'Reviewers_no', 'Review'],
      dtype='object')

### Does it have a weekend?

In [57]:
BangkokData['Has_Weekend'].unique()

array([ True])

In [58]:
BangkokData.drop(['Has_Weekend','Reviewers_no'],axis=1, inplace=True)

### Is Super Host?

In [59]:
BangkokData.Super_host[0:10]

0     Thongjuta is a Superhost
1                          NaN
2          Kiek is a Superhost
3                          NaN
4         Patty is a Superhost
5         Nukul is a Superhost
6                          NaN
7                          NaN
8                          NaN
9    Kinifrog is a Superhost ·
Name: Super_host, dtype: object

In [60]:
BangkokData['Super_host'] = super_host(BangkokData)

In [61]:
BangkokData.Super_host[0:10]

0    1
1    0
2    1
3    0
4    1
5    1
6    0
7    0
8    0
9    1
Name: Super_host, dtype: int64

### Price

In [62]:
BangkokData.Price[0:10]

0    $34
1    $48
2    $27
3    $35
4    $34
5    $23
6    $27
7    $51
8    $38
9    $36
Name: Price, dtype: object

In [63]:
BangkokData['Price'] = price_format(BangkokData)

In [64]:
print(type(BangkokData['Price'][0]))
print(BangkokData['Price'][0:10])

<class 'numpy.float64'>
0    34.0
1    48.0
2    27.0
3    35.0
4    34.0
5    23.0
6    27.0
7    51.0
8    38.0
9    36.0
Name: Price, dtype: float64


### Number of Bathrooms

In [65]:
print(type(BangkokData['Bathrooms_num'][0]))
print(BangkokData['Bathrooms_num'].unique())

<class 'str'>
['1 bath' '1 private bath' '2 baths' '1.5 baths' '0 baths' '1 shared bath'
 '2 shared baths' '4 shared baths' '1.5 private baths' '8 shared baths'
 '1.5 shared baths' '3 baths' '4 baths' '2.5 baths' '3.5 baths' nan
 '4.5 baths'
 'Awesome place! Laundry service was available and affordable, a well equipped kitchen was provided, and the bathrooms and rooms were clean.'
 '6 shared baths' '4.5 shared baths']


In [66]:
BangkokData['Bathrooms_num_int']= float(0)

In [67]:
BangkokData[BangkokData['Bathrooms_num']== 'Awesome place! Laundry service was available and affordable, a well equipped kitchen was provided, and the bathrooms and rooms were clean.']

CheckIn         CheckOut  \
4028  2019-09-10 0:00  2019-09-15 0:00   
4330  2019-09-24 0:00  2019-09-29 0:00   
4609  2019-10-08 0:00  2019-10-13 0:00   
4919  2019-10-22 0:00  2019-10-27 0:00   
5202  2019-11-05 0:00  2019-11-10 0:00   
5207  2019-11-05 0:00  2019-11-10 0:00   
5519  2019-11-19 0:00  2019-11-24 0:00   
5786  2019-12-03 0:00  2019-12-08 0:00   
6079  2019-12-17 0:00  2019-12-22 0:00   
6338  2019-12-31 0:00  2020-01-05 0:00   
6697  2020-01-14 0:00  2020-01-19 0:00   
7296  2020-02-11 0:00  2020-02-16 0:00   
7604  2020-02-25 0:00  2020-03-01 0:00   

                                                  Title  Price  Super_host  \
4028  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
4330  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
4609  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
4919  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
5202  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
5207  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
5519  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
5786  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
6079  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
6338  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
6697  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
7296  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   
7604  Naratel Hostel Bangkok(Bunkbed in 6 bed MixedD...   13.0           1   

      Latitude  Longitude       EntireHomeVsRoom Bedrooms_no Beds_no  \
4028    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
4330    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
4609    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
4919    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
5202    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
5207    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
5519    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
5786    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
6079    13.728    100.527  SHARED ROOM IN HOSTEL   1 bedroom     NaN   
6338    13.728    100.527                    NaN   1 bedroom     NaN   
6697    13.728    100.527                    NaN   1 bedroom     NaN   
7296    13.728    100.527                    NaN   1 bedroom     NaN   
7604    13.728    100.527                    NaN   1 bedroom     NaN   

            ...             Sleeping_engagment  \
4028        ...         Bedroom 1\n3 bunk beds   
4330        ...         Bedroom 1\n3 bunk beds   
4609        ...         Bedroom 1\n3 bunk beds   
4919        ...         Bedroom 1\n3 bunk beds   
5202        ...         Bedroom 1\n3 bunk beds   
5207        ...         Bedroom 1\n3 bunk beds   
5519        ...         Bedroom 1\n3 bunk beds   
5786        ...         Bedroom 1\n3 bunk beds   
6079        ...         Bedroom 1\n3 bunk beds   
6338        ...         Bedroom 1\n3 bunk beds   
6697        ...         Bedroom 1\n3 bunk beds   
7296        ...         Bedroom 1\n3 bunk beds   
7604        ...         Bedroom 1\n3 bunk beds   

                                              Amenities     Accuracy_stars  \
4028  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
4330  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
4609  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
4919  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
5202  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
5207  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5   
5519  ['Wifi', 'Cable TV', 'Laptop friendly workspac...  Rating 5 out of 5  

By looking at the data, I noticed that this sample representing shared room in hostle, and they did not provide any information about the number of bathrooms. For simplicity, I will assume that we have only "1 shared bathroom"! 

In [68]:
for i in range(len(BangkokData)): 
    if(BangkokData['Bathrooms_num'][i]== 'Awesome place! Laundry service was available and affordable, a well equipped kitchen was provided, and the bathrooms and rooms were clean.'):
        BangkokData['Bathrooms_num'][i]= '1 shared bath'
        
    

In [69]:
print(BangkokData['Bathrooms_num'].unique())

['1 bath' '1 private bath' '2 baths' '1.5 baths' '0 baths' '1 shared bath'
 '2 shared baths' '4 shared baths' '1.5 private baths' '8 shared baths'
 '1.5 shared baths' '3 baths' '4 baths' '2.5 baths' '3.5 baths' nan
 '4.5 baths' '6 shared baths' '4.5 shared baths']


In [70]:
BangkokData['Bathrooms_num_int'] = bathrooms_int(BangkokData)

In [71]:
for i in range(10):
    print(BangkokData['Bathrooms_num'][i],BangkokData['Bathrooms_num_int'][i])

1 bath 1.0
1 bath 1.0
1 bath 1.0
1 bath 1.0
1 bath 1.0
1 bath 1.0
1 private bath 1.0
1 bath 1.0
1 bath 1.0
1 bath 1.0


### Shared vs. Private Bathroom

In [72]:
BangkokData['shared_bathroom']= shared_bathroom(BangkokData)

In [214]:
BangkokData[['Bathrooms_num','shared_bathroom']][0:10]

Bathrooms_num  shared_bathroom
0          1 bath            False
1          1 bath            False
2          1 bath            False
3          1 bath            False
4          1 bath            False
5          1 bath            False
6  1 private bath            False
7          1 bath            False
8          1 bath            False
9          1 bath            False

In [215]:
BangkokData.shared_bathroom  = [int(i == True) for i in BangkokData.shared_bathroom]


### Number of Bedrooms 

In [74]:
print(type(BangkokData['Bedrooms_no'][0]))
print(BangkokData['Bedrooms_no'].unique())

<class 'str'>
['1 bedroom' nan
 'Studio accommodation with nice facilities is yours. Facilities are as follows: One king bedroom 2-3 walking distance to Thong Lo Station station'
 'We were very happy when we first booked with Jackrit as the condo had what we were looking for. However, the toilet was an utter disappointment. It had this strong, unbearable odor that permeated through to the bedroom. Even though Jackrit had someone look at it, sadly, it did…Read more'
 'G2015 Single bedroom E1 at Sukhumvit soi64.near sky trian (BTS) Punnawithi station.Room size 26sqm.Have 1 kitchen,1 bathroom and 1 balcony.'
 "Its a nice condo about 12 mins walk to the ON NUT BTS, the condo is on the 8th top floor with good views on BKK city and the pool / condo courtyard,I stayed in another condo after this and whist the other was a one bedroom unit instead of a studio Ho's place and support was…Read more"
 '2 bedrooms' 'New & modern one-bedroom suite near BTS Phyathai'
 'Clean Cozy Condo 1bedroom BTS ba

In [75]:
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('360°view2bedrooms','360° view 2 bedrooms')
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('one-bedroom','1 bedroom')
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('1bedroom','1 bedroom')
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('two bedroom','2 bedrooms')
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('The Zen bedroom','5 bedrooms')
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('One bedroom','1 bedroom')
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('one bedroom','1 bedroom')
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('2bedrooms','2 bedrooms')
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('4-bedrooms','4 bedrooms')
BangkokData.Bedrooms_no = BangkokData.Bedrooms_no.str.replace('3-bedrooms','3 bedrooms')


In [76]:
BangkokData['Bedrooms_int']= None 

In [77]:
BangkokData['Bedrooms_int'] = bedrooms_int(BangkokData)

In [78]:
BangkokData['Bedrooms_int'].unique()

array(['1', -1, 'king', 'the', 'Single', '2', 'Big', 'King', 'pool-Comfy',
       'size', '5', 'BTS/1', 'from', '3', '4', '6', '☼Sky'], dtype=object)

In [79]:
oneList = ['king', 'the', 'Single', 'Big', 'King','pool-Comfy', 'size', 'BTS/1', 'from', '☼Sky']
for i in oneList:
    BangkokData['Bedrooms_int']= BangkokData['Bedrooms_int'].replace(i,'1')


In [80]:
BangkokData['Bedrooms_int'].unique()

array(['1', -1, '2', '5', '3', '4', '6'], dtype=object)

In [81]:
BangkokData['Bedrooms_int']= bedrooms_int_final(BangkokData)

In [84]:
BangkokData['Bedrooms_int'].head(10)

0     1
1     1
2     1
3     1
4    -1
5     1
6    -1
7     1
8     1
9     1
Name: Bedrooms_int, dtype: object

### Number of Beds

In [85]:
print(type(BangkokData['Beds_no'][0]))
print(BangkokData['Beds_no'].unique())

<class 'str'>
['1 bed' '2 beds' '3 beds' nan '4 beds' '0 beds'
 'PRIVATE ROOM IN BED AND BREAKFAST' '5 beds' '6 beds' '7 beds' '8 beds'
 'Private room in bed and breakfast']


In [86]:
BangkokData['Beds_int'] = float(0)

In [87]:
BangkokData['Beds_int']= beds_int(BangkokData)

In [88]:
for i in range(10):
    print(BangkokData['Beds_no'][i],BangkokData['Beds_int'][i])

1 bed 1.0
1 bed 1.0
2 beds 2.0
1 bed 1.0
3 beds 3.0
3 beds 3.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
1 bed 1.0


### Latitude

In [89]:
print(type(BangkokData['Latitude'][0]))

<class 'numpy.float64'>


### Longitude

In [90]:
print(type(BangkokData['Longitude'][0]))

<class 'numpy.float64'>


# Cairo

In [91]:
CairoData.columns

Index(['CheckIn', 'CheckOut', 'Has_Weekend', 'Title', 'Price', 'Super_host',
       'Latitude', 'Longitude', 'EntireHomeVsRoom', 'Bedrooms_no', 'Beds_no',
       'Bathrooms_num', 'Sleeping_engagment', 'Amenities', 'Accuracy_stars',
       'Communication_stars', 'Cleanliness_stars', 'Location_stars',
       'CheckIn_stars', 'Value_stars', 'Reviewers_no', 'Review'],
      dtype='object')

### Does it have a weekend?

In [92]:
CairoData['Has_Weekend'].unique()

array([ True])

In [93]:
CairoData.drop(['Has_Weekend','Reviewers_no'],axis=1, inplace=True)

In [94]:
CairoData.columns

Index(['CheckIn', 'CheckOut', 'Title', 'Price', 'Super_host', 'Latitude',
       'Longitude', 'EntireHomeVsRoom', 'Bedrooms_no', 'Beds_no',
       'Bathrooms_num', 'Sleeping_engagment', 'Amenities', 'Accuracy_stars',
       'Communication_stars', 'Cleanliness_stars', 'Location_stars',
       'CheckIn_stars', 'Value_stars', 'Review'],
      dtype='object')

### Is Super Host?

In [95]:
CairoData.Super_host[0:10]

0                       NaN
1                       NaN
2                       NaN
3                       NaN
4                       NaN
5                       NaN
6                       NaN
7     Bassem is a Superhost
8                       NaN
9    Mohamed is a Superhost
Name: Super_host, dtype: object

In [96]:
CairoData['Super_host'] = super_host(CairoData)

In [97]:
CairoData.Super_host[0:10]

0    0
1    0
2    0
3    0
4    0
5    0
6    0
7    1
8    0
9    1
Name: Super_host, dtype: int64

### Price

In [98]:
CairoData['Price'] = price_format(CairoData)

In [99]:
print(type(CairoData['Price'][0]))
print(CairoData['Price'][0:10])

<class 'numpy.float64'>
0    15.0
1    12.0
2    22.0
3    19.0
4    23.0
5    25.0
6    40.0
7    37.0
8    65.0
9    11.0
Name: Price, dtype: float64


### Number of Bathrooms

In [100]:
print(type(CairoData['Bathrooms_num'][0]))
print(CairoData['Bathrooms_num'].unique())

<class 'str'>
['2 shared baths' '1.5 baths' '1 shared bath' '1.5 private baths'
 '1 private bath' '1 bath' '2 baths' '3 baths' '12 private baths'
 '8 private baths' '1.5 shared baths' '2.5 baths' nan '4 baths'
 '2 private baths' '0 shared baths' '3 private baths' '3 shared baths'
 '10 private baths' '5 private baths' '6 baths' '4 private baths'
 '3.5 baths' '4 shared baths' '25 private baths' '20 private baths']


In [101]:
CairoData['Bathrooms_num_int']= float(0)

In [102]:
CairoData['Bathrooms_num_int']= bathrooms_int(CairoData) 

In [103]:
for i in range(10):
    print(CairoData['Bathrooms_num'][i],CairoData['Bathrooms_num_int'][i])

2 shared baths 2.0
1.5 baths 1.5
1 shared bath 1.0
1.5 private baths 1.5
1 shared bath 1.0
1 private bath 1.0
1 private bath 1.0
1.5 baths 1.5
1 bath 1.0
1 shared bath 1.0


### Shared vs. Private Bathrooms

In [104]:
CairoData['shared_bathroom']= shared_bathroom(CairoData)

In [105]:
CairoData[['Bathrooms_num','shared_bathroom']][0:10]

Bathrooms_num  shared_bathroom
0     2 shared baths             True
1          1.5 baths            False
2      1 shared bath             True
3  1.5 private baths            False
4      1 shared bath             True
5     1 private bath            False
6     1 private bath            False
7          1.5 baths            False
8             1 bath            False
9      1 shared bath             True

In [217]:
CairoData.shared_bathroom  = [int(i == True) for i in CairoData.shared_bathroom]


### Number of Bedrooms

In [106]:
print(type(CairoData['Bedrooms_no'][0]))
print(CairoData['Bedrooms_no'].unique())

<class 'str'>
['1 bedroom' '2 bedrooms'
 'This fully furnished studio garden apartment includes 1 bedroom, 2 reception units, 1 bathroom..'
 'Expat apartment - Clean and cosy bedroom' nan
 'Cairo Zahraa El Maadi, one bedroom.'
 'Paradise at the Point! This luxurious 1 bedroom, 1 bathroom bath modern-built estate on the Open view is a dream home in a high-end community.'
 '20 bedrooms' '3 bedrooms'
 'Double bedroom in Paris Hotel Breakfast included .'
 'Cozy queen bedroom, perfect location.'
 'Cozy,clean apartment located in Maadi, one bedroom'
 '2 Single bedroom in Paris Hotel Breakfast included'
 'Apartment(3bedroom) Eltagamo new cairo - 4' '4 bedrooms'
 'Private bedroom in a cosy apartment' '10 bedrooms'
 'Apartment(2bedroom) Eltagamo new cairo'
 'One bedroom with garden in The Village, New Cairo.' '6 bedrooms'
 '15 bedrooms' 'Apartment(3bedroom) Eltagamo new cairo - 5'
 'Apartment(3bedroom) Eltagamo new cairo - 3'
 'Princess Apartment 3 bedrooms' 'Beautiful 2 bedrooms close to down 

In [107]:
CairoData['Bedrooms_no'] = CairoData['Bedrooms_no'].str.replace('2 Single bedroom','2 bedroom')
CairoData['Bedrooms_no'] = CairoData['Bedrooms_no'].str.replace('3bedroom','3 bedroom')
CairoData['Bedrooms_no'] = CairoData['Bedrooms_no'].str.replace('2bedroom','2 bedroom')
CairoData['Bedrooms_no'] = CairoData['Bedrooms_no'].str.replace('2-bedroom','2 bedroom')
CairoData['Bedrooms_no'] = CairoData['Bedrooms_no'].str.replace('BOTH (2) comfortable DBL bedrooms','2 bedroom')
CairoData['Bedrooms_no'] = CairoData['Bedrooms_no'].str.replace('one bedroom','1 bedroom')
CairoData['Bedrooms_no'] = CairoData['Bedrooms_no'].str.replace('One bedroom','1 bedroom')
CairoData['Bedrooms_no'] = CairoData['Bedrooms_no'].str.replace('Three bedroom','3 bedroom')
CairoData['Bedrooms_no'] = CairoData['Bedrooms_no'].str.replace('Two bedroom','3 bedroom')


In [108]:
CairoData['Bedrooms_int']= None 

In [109]:
CairoData['Bedrooms_int']= bedrooms_int(CairoData)

In [110]:
CairoData['Bedrooms_int'].unique()

array(['1', '2', 'cosy', -1, '20', '3', 'Double', 'queen', 'Apartment(3',
       '4', 'Private', '10', 'Apartment(2', '6', '15', '☀☼Sunny', 'Guest',
       'nice', '5', '13', '9', 'DBL', '25', 'Cozy', 'Master'],
      dtype=object)

In [111]:
oneList = ['cosy','Double', 'queen','Private','☀☼Sunny','Guest','nice','DBL','Cozy','Master']
for i in oneList:
    CairoData['Bedrooms_int']= CairoData['Bedrooms_int'].replace(i,'1')
    
CairoData['Bedrooms_int']= CairoData['Bedrooms_int'].replace('Apartment(3','3')
CairoData['Bedrooms_int']= CairoData['Bedrooms_int'].replace('Apartment(2','2')

In [112]:
CairoData['Bedrooms_int'].unique()

array(['1', '2', -1, '20', '3', '4', '10', '6', '15', '5', '13', '9',
       '25'], dtype=object)

In [113]:
CairoData['Bedrooms_int'] = bedrooms_int_final(CairoData)

In [114]:
CairoData['Bedrooms_int'].unique()

array(['1', '2', -1, '20', '3', '4', '10', '6', '15', '5', '13', '9',
       '25'], dtype=object)

### Number of Beds

In [115]:
print(type(CairoData['Beds_no'][0]))
print(CairoData['Beds_no'].unique())

<class 'str'>
['2 beds' '1 bed' '4 beds' nan 'Private room in bed and breakfast'
 '3 beds' '26 beds' '6 beds' '5 beds' '8 beds'
 'Stylish brand new fully furnished studio apartment with queen size bed and open plan kitchen. kitchen is fully equipped with all cutlery and crockery, washing machine, cooker, microwave, kettle, toaster. bed linen and towels are provided.'
 '16 beds' '50 beds' '37 beds' '7 beds' '10 beds' '18 beds' '30 beds']


In [116]:
CairoData[CairoData['Beds_no']== 'Stylish brand new fully furnished studio apartment with queen size bed and open plan kitchen. kitchen is fully equipped with all cutlery and crockery, washing machine, cooker, microwave, kettle, toaster. bed linen and towels are provided.']

CheckIn         CheckOut  \
70    2019-03-14 0:00  2019-03-19 0:00   
557   2019-03-28 0:00  2019-04-02 0:00   
896   2019-04-11 0:00  2019-04-16 0:00   
902   2019-04-11 0:00  2019-04-16 0:00   
1211  2019-04-25 0:00  2019-04-30 0:00   
2102  2019-06-06 0:00  2019-06-11 0:00   
2380  2019-06-20 0:00  2019-06-25 0:00   
2681  2019-07-04 0:00  2019-07-09 0:00   
3001  2019-07-18 0:00  2019-07-23 0:00   
3317  2019-08-01 0:00  2019-08-06 0:00   
3633  2019-08-15 0:00  2019-08-20 0:00   
3920  2019-08-29 0:00  2019-09-03 0:00   
4169  2019-09-12 0:00  2019-09-17 0:00   
4784  2019-10-10 0:00  2019-10-15 0:00   
5095  2019-10-24 0:00  2019-10-29 0:00   
5309  2019-11-07 0:00  2019-11-12 0:00   
5614  2019-11-21 0:00  2019-11-26 0:00   
5917  2019-12-05 0:00  2019-12-10 0:00   
6212  2019-12-19 0:00  2019-12-24 0:00   
6510  2020-01-02 0:00  2020-01-07 0:00   
6816  2020-01-16 0:00  2020-01-21 0:00   
7119  2020-01-30 0:00  2020-02-04 0:00   
7423  2020-02-13 0:00  2020-02-18 0:00   
7725  2020-02-27 0:00  2020-03-03 0:00   

                                                  Title  Price  Super_host  \
70    fully furnished apt in the village palm hills,...   63.0           0   
557   fully furnished apt in the village palm hills,...   63.0           0   
896   fully furnished apt in the village palm hills,...   63.0           0   
902   fully furnished apt in the village palm hills,...   63.0           0   
1211  fully furnished apt in the village palm hills,...   63.0           0   
2102  fully furnished apt in the village palm hills,...   63.0           0   
2380  fully furnished apt in the village palm hills,...   63.0           0   
2681  fully furnished apt in the village palm hills,...   63.0           0   
3001  fully furnished apt in the village palm hills,...   63.0           0   
3317  fully furnished apt in the village palm hills,...   69.0           0   
3633  fully furnished apt in the village palm hills,...   70.0           0   
3920  fully furnished apt in the village palm hills,...   71.0           0   
4169  fully furnished apt in the village palm hills,...   77.0           0   
4784  fully furnished apt in the village palm hills,...   78.0           0   
5095  fully furnished apt in the village palm hills,...   78.0           0   
5309  fully furnished apt in the village palm hills,...   78.0           0   
5614  fully furnished apt in the village palm hills,...   78.0           0   
5917  fully furnished apt in the village palm hills,...   78.0           0   
6212  fully furnished apt in the village palm hills,...   78.0           0   
6510  fully furnished apt in the village palm hills,...   78.0           0   
6816  fully furnished apt in the village palm hills,...   78.0           0   
7119  fully furnished apt in the village palm hills,...   78.0           0   
7423  fully furnished apt in the village palm hills,...   78.0           0   
7725  fully furnished apt in the village palm hills,...    NaN           0   

      Latitude  Longitude  EntireHomeVsRoom Bedrooms_no  \
70    30.01949   31.50053  Entire apartment   1 bedroom   
557   30.01949   31.50053  Entire apartment   1 bedroom   
896   30.01949   31.50053  Entire apartment   1 bedroom   
902   30.01949   31.50053  Entire apartment   1 bedroom   
1211  30.01949   31.50053  Entire apartment   1 bedroom   
2102  30.01949   31.50053  Entire apartment   1 bedroom   
2380  30.01949   31.50053  Entire apartment   1 bedroom   
2681  30.01949   31.50053  Entire apartment   1 bedroom   
3001  30.01949   31.50053  Entire apartment   1 bedroom   
3317  30.01949   31.50053  Entire apartment   1 bedroom   
3633  30.01949   31.50053  Entire apartment   1 bedroom   
3920  30.01949   31.50053  Entire apartment   1 bedroom   
4169  30.01949   31.50053  Entire apartment   1 bedroom   
4784  30.01949   31.50053  Entire apartment   1 bedroom   
5095  30.01949   31.50053  Entire apartment   1 bedroom   
5309  30.01949   31.50053  Entire apartment   1 bedroom   
5614  30.

In [117]:
for i in range(len(BangkokData)): 
    if(CairoData['Beds_no'][i]== 'Stylish brand new fully furnished studio apartment with queen size bed and open plan kitchen. kitchen is fully equipped with all cutlery and crockery, washing machine, cooker, microwave, kettle, toaster. bed linen and towels are provided.'):
        CairoData['Beds_no'][i]= '1 bed'

In [118]:
CairoData['Beds_int']= float(0)

In [119]:
CairoData['Beds_int']= beds_int(CairoData)

In [120]:
for i in range(10):
    print(CairoData['Beds_no'][i],CairoData['Beds_int'][i])

2 beds 2.0
1 bed 1.0
1 bed 1.0
2 beds 2.0
1 bed 1.0
1 bed 1.0
1 bed 1.0
2 beds 2.0
1 bed 1.0
1 bed 1.0


### Latitude

In [121]:
print(type(CairoData['Latitude'][0]))

<class 'numpy.float64'>


### Longitude

In [122]:
print(type(CairoData['Longitude'][0]))

<class 'numpy.float64'>


## Categorical Data

### Review Stars

In [123]:
stars_list =['Accuracy_stars','CheckIn_stars','Cleanliness_stars','Communication_stars','Location_stars','Value_stars']

### NewYork

In [124]:
for i in stars_list:
    print(NewYorkData[i].unique())


['Rating 5 out of 5' 'Rating 4.5 out of 5' nan 'Rating 4 out of 5'
 'Rating 2.5 out of 5' 'Rating 3.5 out of 5']
['Rating 5 out of 5' 'Rating 4.5 out of 5' nan 'Rating 4 out of 5'
 'Rating 3.5 out of 5' 'Rating 2 out of 5']
['Rating 5 out of 5' 'Rating 4 out of 5' nan 'Rating 3.5 out of 5'
 'Rating 4.5 out of 5' 'Rating 3 out of 5' 'Rating 2.5 out of 5']
['Rating 5 out of 5' 'Rating 4.5 out of 5' nan 'Rating 4 out of 5'
 'Rating 3.5 out of 5' 'Rating 2 out of 5']
['Rating 5 out of 5' 'Rating 4.5 out of 5' nan 'Rating 4 out of 5'
 'Rating 3.5 out of 5']
['Rating 4.5 out of 5' 'Rating 4 out of 5' 'Rating 5 out of 5' nan
 'Rating 2 out of 5' 'Rating 3.5 out of 5']


In [125]:
num_of_stars(NewYorkData)

Accuracy_stars                                          Amenities  \
0                 5  ['Indoor fireplace', 'Wifi', 'Iron', 'Laptop f...   
1               4.5  ['Indoor fireplace', 'Wifi', 'Laptop friendly ...   
2                 5  ['Wifi', 'Essentials', 'Heating', 'Air conditi...   
3                 5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
4                 5  ['Wifi', 'Washer', 'Dryer', 'Iron', 'Laptop fr...   
5                 0  ['Wifi', 'Essentials', 'Heating', 'Air conditi...   
6                 5  ['Wifi', 'Washer', 'Dryer', 'Iron', 'Laptop fr...   
7                 5  ['Wifi', 'Washer', 'Dryer', 'Iron', 'Laptop fr...   
8               4.5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
9                 5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
10                0  ['Wifi', 'TV', 'Essentials', 'Heating', 'Air c...   
11                0  ['Wifi', 'TV', 'Essentials', 'Heating', 'Priva...   
12                5  ['Wifi', 'Iron', 'TV', 'Essentials', 'Air cond...   
13                5  ['Wifi', 'Laptop friendly workspace', 'TV', 'E...   
14                5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
15                0  ['Wifi', 'Washer', 'Dryer', 'Iron', 'Laptop fr...   
16                5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
17                0  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
18                5  ['Wifi', 'Essentials', 'Heating', 'Hot water',...   
19                0  ['Wifi', 'Iron', 'Essentials', 'Heating', 'Air...   
20                5  ['Wifi', 'Washer', 'Dryer', 'Iron', 'Laptop fr...   
21                5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
22              4.5  ['Wifi', 'Cable TV', 'Washer', 'Dryer', 'TV', ...   
23                5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
24                0  ['Wifi', 'Iron', 'TV', 'Essentials', 'Heating'...   
25                0  ['Wifi', 'Iron', 'Essentials', 'Heating', 'Air...   
26                5  ['Wifi', 'Cable TV', 'Iron', 'Laptop friendly ...   
27              4.5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
28                5  ['Wifi', 'Cable TV', 'Washer', 'Dryer', 'Iron'...   
29                5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
...             ...                                                ...   
3927              5  ['Wifi', 'Cable TV', 'Washer', 'Dryer', 'Iron'...   
3928              5  ['Wifi', 'Cable TV', 'Iron', 'Laptop friendly ...   
3929              5  ['Wifi', 'Cable TV', 'Washer', 'Dryer', 'Iron'...   
3930              5  ['Wifi', 'Cable TV', 'Dryer', 'Iron', 'Laptop ...   
3931              5  ['Wifi', 'Cable TV', 'Washer', 'Dryer', 'Iron'...   
3932              5  ['Wifi', 'Washer', 'Dryer', 'Iron', 'Laptop fr...   
3933              5  ['Wifi', 'Cable TV', 'Iron', 'Laptop friendly ...   
3934              5  ['Wifi', 'Cable TV', 'Washer', 'Dryer', 'Iron'...   
3935            4.5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
3936              5  ['Indoor fireplace', 'Wifi', 'Iron', 'Laptop f...   
3937              5  ['Wifi', 'Essentials', 'Heating', 'Air conditi...   
3938            4.5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
3939              5  ['Wifi', 'Cable TV', 'Washer', 'Dryer', 'Iron'...   
3940              5  ['Wifi', 'Cable TV', 'Essentials', 'Heating', ...   
3941            4.5  ['Wifi', 'Cable TV', 'Iron', 'Laptop friendly ...   
3942              5  ['Wifi', 'Cable TV', 'Iron', 'Laptop friendly ...   
3943              5  ['Wifi', 'Cable TV', 'Iron', 'Laptop friendly ...   
3944            4.5  ['Wifi', 'Cable TV', 'Iron', 'Laptop friendly ...   
3945              5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
3946              5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
3947              5  ['Wifi', 'Cable TV', 'Washer', 'Dryer', 'Iron'...   
3948              5  ['Wifi', 'Iron', 'Laptop friendly workspace', ...   
3949     

In [126]:
for i in stars_list:
    print(NewYorkData[i].unique())


['5' '4.5' '0' '4' '2.5' '3.5']
['5' '4.5' '0' '4' '3.5' '2']
['5' '4' '0' '3.5' '4.5' '3' '2.5']
['5' '4.5' '0' '4' '3.5' '2']
['5' '4.5' '0' '4' '3.5']
['4.5' '4' '5' '0' '2' '3.5']


### Entire Apartment vs. Shared Room vs. Private Room

In [127]:
NewYorkData.EntireHomeVsRoom.unique()

array(['ENTIRE APARTMENT', 'PRIVATE ROOM IN APARTMENT',
       'PRIVATE ROOM IN HOUSE', 'PRIVATE ROOM IN LOFT',
       'PRIVATE ROOM IN TOWNHOUSE', 'PRIVATE ROOM IN BED AND BREAKFAST',
       'SHARED ROOM IN APARTMENT', 'PRIVATE ROOM IN CONDOMINIUM',
       'PRIVATE ROOM IN SERVICED APARTMENT',
       'PRIVATE ROOM IN GUEST SUITE', 'PRIVATE ROOM IN HOSTEL',
       'ENTIRE LOFT', 'ENTIRE TOWNHOUSE', 'SHARED ROOM IN LOFT',
       'ENTIRE HOUSE', 'SHARED ROOM IN APARTHOTEL',
       'ENTIRE SERVICED APARTMENT', 'ENTIRE CONDOMINIUM', 'PRIVATE ROOM',
       'ROOM IN BOUTIQUE HOTEL', 'SHARED ROOM IN BED AND BREAKFAST',
       'PRIVATE ROOM IN GUESTHOUSE', 'ENTIRE GUEST SUITE',
       'PRIVATE ROOM IN RESORT', 'ENTIRE GUESTHOUSE',
       'SHARED ROOM IN HOUSE', 'PRIVATE ROOM IN TINY HOUSE',
       'ENTIRE BED AND BREAKFAST'], dtype=object)

In [128]:
## Entire apartment = 0
## Shared room = 1
## Private room = 2
NewYorkData['shared_room']= int(0)

In [130]:
NewYorkData['EntireHomeVsRoom'].head(10)

0             ENTIRE APARTMENT
1    PRIVATE ROOM IN APARTMENT
2        PRIVATE ROOM IN HOUSE
3    PRIVATE ROOM IN APARTMENT
4    PRIVATE ROOM IN APARTMENT
5    PRIVATE ROOM IN APARTMENT
6         PRIVATE ROOM IN LOFT
7    PRIVATE ROOM IN APARTMENT
8    PRIVATE ROOM IN APARTMENT
9    PRIVATE ROOM IN TOWNHOUSE
Name: EntireHomeVsRoom, dtype: object

In [131]:
NewYorkData['shared_room']= entire_shared_private_room(NewYorkData)

In [132]:
for i in range(20):
    print(NewYorkData.EntireHomeVsRoom[i],NewYorkData['shared_room'][i])

ENTIRE APARTMENT 0
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN HOUSE 2
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN LOFT 2
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN TOWNHOUSE 2
ENTIRE APARTMENT 0
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN HOUSE 2
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN BED AND BREAKFAST 2
ENTIRE APARTMENT 0
SHARED ROOM IN APARTMENT 1
PRIVATE ROOM IN APARTMENT 2
PRIVATE ROOM IN APARTMENT 2


### Sleeping Engagment

In [133]:
for i in NewYorkData.Sleeping_engagment:
    if(str(i)!='nan'):
        if(i.split(' ')[2]=='small'):
            print(i)

Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed


In [134]:
NewYorkData.Sleeping_engagment = NewYorkData.Sleeping_engagment.str.replace('small double','double')

In [135]:
NewYorkData['BedType']= None
NewYorkData['BedType'] = bed_type(NewYorkData)
    

In [136]:
NewYorkData['BedType'].unique()

array(['queen', 'single', None, 'king', 'double', 'bunk', 'couch', 'sofa',
       'couches', 'couch,', 'crib', 'air', 'floor'], dtype=object)

In [140]:
NewYorkData = one_hot_encode_bedtype(NewYorkData)

In [142]:
NewYorkData.columns


Index(['Accuracy_stars', 'Amenities', 'Bathrooms_num', 'Bedrooms_no',
       'Beds_no', 'CheckIn', 'CheckIn_stars', 'CheckOut', 'Cleanliness_stars',
       'Communication_stars', 'EntireHomeVsRoom', 'Latitude', 'Location_stars',
       'Longitude', 'Price', 'Review', 'Sleeping_engagment', 'Super_host',
       'Title', 'Value_stars', 'Bathrooms_num_int', 'shared_bathroom',
       'Bedrooms_int', 'Beds_int', 'shared_room', 'BedType', 'air', 'bunk',
       'couch', 'couch,', 'couches', 'crib', 'double', 'floor', 'king',
       'queen', 'single', 'sofa'],
      dtype='object')

In [148]:
NewYorkData.drop(['BedType','couch,', 'couches'],axis=1, inplace=True)

### Amenities

In [149]:
NewYorkData['Amenities_mod']= None

In [150]:
NewYorkData['Amenities_mod']= amenities_to_vars(NewYorkData)

In [151]:
NewYorkData.Amenities_mod[0:30]

0     [Indoor fireplace, Wifi, Iron, Laptop friendly...
1     [Indoor fireplace, Wifi, Laptop friendly works...
2     [Wifi, Essentials, Heating, Air conditioning, ...
3     [Wifi, Iron, Laptop friendly workspace, Essent...
4     [Wifi, Washer, Dryer, Iron, Laptop friendly wo...
5     [Wifi, Essentials, Heating, Air conditioning, ...
6     [Wifi, Washer, Dryer, Iron, Laptop friendly wo...
7     [Wifi, Washer, Dryer, Iron, Laptop friendly wo...
8     [Wifi, Iron, Laptop friendly workspace, Essent...
9     [Wifi, Iron, Laptop friendly workspace, Essent...
10    [Wifi, TV, Essentials, Heating, Air conditioni...
11    [Wifi, TV, Essentials, Heating, Private entran...
12    [Wifi, Iron, TV, Essentials, Air conditioning,...
13    [Wifi, Laptop friendly workspace, TV, Essentia...
14    [Wifi, Iron, Laptop friendly workspace, Essent...
15    [Wifi, Washer, Dryer, Iron, Laptop friendly wo...
16    [Wifi, Iron, Laptop friendly workspace, TV, Es...
17    [Wifi, Iron, Laptop friendly workspace, Es

In [152]:
len(NewYorkData.Amenities_mod[0])

28

In [153]:
NewYorkData.Amenities_mod[0]

['Indoor fireplace',
 'Wifi',
 'Iron',
 'Laptop friendly workspace',
 'TV',
 'Essentials',
 'Heating',
 'Air conditioning',
 'Hot water',
 'Bathtub',
 'Fireplace guards',
 'Kitchen',
 'Cooking basics',
 'Dishes and silverware',
 'Dishwasher',
 'Microwave',
 'Refrigerator',
 'Oven',
 'Stove',
 'Lockbox',
 'Luggage dropoff allowed',
 'Long term stays allowed',
 'Hangers',
 'Hair dryer',
 'Shampoo',
 'Patio or balcony',
 'Carbon monoxide detector',
 'Smoke detector']

In [193]:
NewYorkData=encode_amenities(NewYorkData)

In [194]:
NewYorkData.columns

Index(['Accuracy_stars', 'Amenities', 'Bathrooms_num', 'Bedrooms_no',
       'Beds_no', 'CheckIn', 'CheckIn_stars', 'CheckOut', 'Cleanliness_stars',
       'Communication_stars',
       ...
       'Smart lock', 'Smoke detector', 'Stair gates', 'Stove', 'TV',
       'Table corner guards', 'Washer', 'Waterfront', 'Wifi', 'Window guards'],
      dtype='object', length=115)

In [198]:
NewYorkData.drop(['Amenities'],axis=1, inplace = True)

In [219]:
NewYorkData.drop(['Title','EntireHomeVsRoom','Bedrooms_no','Beds_no','Bathrooms_num','Sleeping_engagment','Review'],axis=1 ,inplace = True)

### Bangkok

### Review Stars

In [154]:
for i in stars_list:
    print(BangkokData[i])

0         Rating 5 out of 5
1                       NaN
2         Rating 5 out of 5
3         Rating 5 out of 5
4                       NaN
5         Rating 5 out of 5
6                       NaN
7         Rating 5 out of 5
8                       NaN
9         Rating 5 out of 5
10                      NaN
11        Rating 5 out of 5
12        Rating 5 out of 5
13      Rating 4.5 out of 5
14                      NaN
15        Rating 5 out of 5
16                      NaN
17                      NaN
18                      NaN
19                      NaN
20        Rating 4 out of 5
21        Rating 5 out of 5
22        Rating 5 out of 5
23        Rating 5 out of 5
24      Rating 4.5 out of 5
25        Rating 5 out of 5
26        Rating 5 out of 5
27        Rating 5 out of 5
28      Rating 4.5 out of 5
29      Rating 4.5 out of 5
               ...         
7721      Rating 5 out of 5
7722      Rating 5 out of 5
7723      Rating 5 out of 5
7724    Rating 4.5 out of 5
7725    Rating 4.5 o

In [155]:
num_of_stars(BangkokData)

CheckIn         CheckOut  \
0     2019-03-12 0:00  2019-03-17 0:00   
1     2019-03-12 0:00  2019-03-17 0:00   
2     2019-03-12 0:00  2019-03-17 0:00   
3     2019-03-12 0:00  2019-03-17 0:00   
4     2019-03-12 0:00  2019-03-17 0:00   
5     2019-03-12 0:00  2019-03-17 0:00   
6     2019-03-12 0:00  2019-03-17 0:00   
7     2019-03-12 0:00  2019-03-17 0:00   
8     2019-03-12 0:00  2019-03-17 0:00   
9     2019-03-12 0:00  2019-03-17 0:00   
10    2019-03-12 0:00  2019-03-17 0:00   
11    2019-03-12 0:00  2019-03-17 0:00   
12    2019-03-12 0:00  2019-03-17 0:00   
13    2019-03-12 0:00  2019-03-17 0:00   
14    2019-03-12 0:00  2019-03-17 0:00   
15    2019-03-12 0:00  2019-03-17 0:00   
16    2019-03-12 0:00  2019-03-17 0:00   
17    2019-03-12 0:00  2019-03-17 0:00   
18    2019-03-12 0:00  2019-03-17 0:00   
19    2019-03-12 0:00  2019-03-17 0:00   
20    2019-03-12 0:00  2019-03-17 0:00   
21    2019-03-12 0:00  2019-03-17 0:00   
22    2019-03-12 0:00  2019-03-17 0:00   
23    2019-03-12 0:00  2019-03-17 0:00   
24    2019-03-12 0:00  2019-03-17 0:00   
25    2019-03-12 0:00  2019-03-17 0:00   
26    2019-03-12 0:00  2019-03-17 0:00   
27    2019-03-12 0:00  2019-03-17 0:00   
28    2019-03-12 0:00  2019-03-17 0:00   
29    2019-03-12 0:00  2019-03-17 0:00   
...               ...              ...   
7721  2020-02-25 0:00  2020-03-01 0:00   
7722  2020-02-25 0:00  2020-03-01 0:00   
7723  2020-02-25 0:00  2020-03-01 0:00   
7724  2020-02-25 0:00  2020-03-01 0:00   
7725  2020-02-25 0:00  2020-03-01 0:00   
7726  2020-02-25 0:00  2020-03-01 0:00   
7727  2020-02-25 0:00  2020-03-01 0:00   
7728  2020-02-25 0:00  2020-03-01 0:00   
7729  2020-02-25 0:00  2020-03-01 0:00   
7730  2020-02-25 0:00  2020-03-01 0:00   
7731  2020-02-25 0:00  2020-03-01 0:00   
7732  2020-02-25 0:00  2020-03-01 0:00   
7733  2020-02-25 0:00  2020-03-01 0:00   
7734  2020-02-25 0:00  2020-03-01 0:00   
7735  2020-02-25 0:00  2020-03-01 0:00   
7736  2020-02-25 0:00  2020-03-01 0:00   
7737  2020-02-25 0:00  2020-03-01 0:00   
7738  2020-02-25 0:00  2020-03-01 0:00   
7739  2020-02-25 0:00  2020-03-01 0:00   
7740  2020-02-25 0:00  2020-03-01 0:00   
7741  2020-02-25 0:00  2020-03-01 0:00   
7742  2020-02-25 0:00  2020-03-01 0:00   
7743  2020-02-25 0:00  2020-03-01 0:00   
7744  2020-02-25 0:00  2020-03-01 0:00   
7745  2020-02-25 0:00  2020-03-01 0:00   
7746  2020-02-25 0:00  2020-03-01 0:00   
7747  2020-02-25 0:00  2020-03-01 0:00   
7748  2020-02-25 0:00  2020-03-01 0:00   
7749  2020-02-25 0:00  2020-03-01 0:00   
7750  2020-02-25 0:00  2020-03-01 0:00   

                                                  Title  Price  Super_host  \
0                   Cosy room with 3 mins walk from MRT   34.0           1   
1                                          曼谷市区BTS旁品质公寓   48.0           0   
2     Modern apartment close to city near sky train(...   27.0           1   
3     Outstanding Location in Heart of Bangkok 2mins...   35.0           0   
4     5 Min walk MRT Phraram9, 1stop RodFai Train Ma...   34.0           1   
5      Loft style downtown near MRT 35sqm เช่าราย3เดือน   23.0           1   
6     Start from 599THB !!! Walk to RCA in few steps!!!   27.0           0   
7                LoveBangkok民宿 双人房 3071 市中心/轻轨旁(全套乳胶寝具)   51.0           0   
8     City Home 1 BR Sukhumvit 103 (BTS Udom Suk) BITEC   38.0           0   
9       RiverViewSuiteA1:Netflix+FreeWIFI@room and Pool   36.0           1   
10    BTS 2min 13F看梅兰河夜景大一房一厅big good room-ICONSIAM ...   46.0           1   
11     Bangkok Apartments BTS Thong Lo by walk 3min 308   38.0           0   
12    Warm&Inviting Spacious Studio Condo • BTS 5min...   34.0           0   
13            Green view private room on Sukhumvit Road   25.0           1   
14         One bed room/ wifi / markrt / city /shopping   34.0           0   
15             350m to MRT Bangson - Cozy 1BR apartment   26.0           1   
16           地铁口80米 /安全干净舒适便利/代购街中餐厅夜市丰富/象神佛牌/免费泳池Wi-Fi   33.0           0   
17   

In [156]:
for i in stars_list:
    print(BangkokData[i].unique())

['5' '0' '4.5' '4' '3.5']
['5' '0' '4.5' '4' '3.5' '3']
['4.5' '0' '5' '4' '3.5']
['5' '0' '4.5' '4' '3.5']
['5' '0' '4' '4.5' '3.5' '3']
['5' '0' '4.5' '4' '2.5' '3.5']


### Entire Apartment vs. Shared Room vs. Private Room

In [157]:
BangkokData.EntireHomeVsRoom.unique()

array(['ENTIRE APARTMENT', 'ENTIRE CONDOMINIUM',
       'PRIVATE ROOM IN APARTMENT', 'ENTIRE SERVICED APARTMENT',
       'ROOM IN BOUTIQUE HOTEL', 'ENTIRE LOFT',
       'PRIVATE ROOM IN CONDOMINIUM', 'PRIVATE ROOM IN VILLA',
       'PRIVATE ROOM IN SERVICED APARTMENT', 'PRIVATE ROOM IN TOWNHOUSE',
       'PRIVATE ROOM IN LOFT', 'ENTIRE PLACE', 'ENTIRE TOWNHOUSE',
       'ROOM IN HOTEL', 'ENTIRE GUEST SUITE', 'PRIVATE ROOM IN HOSTEL',
       'ENTIRE HOUSE', 'PRIVATE ROOM IN GUESTHOUSE',
       'PRIVATE ROOM IN BED AND BREAKFAST', 'ENTIRE VILLA',
       'ENTIRE GUESTHOUSE', 'PRIVATE ROOM IN HOUSE', 'ROOM IN APARTHOTEL',
       'SHARED ROOM IN HOSTEL', 'ENTIRE HOSTEL', 'TINY HOUSE',
       'ENTIRE COTTAGE', nan], dtype=object)

In [158]:
## Entire apartment = 0
## Shared room = 1
## Private room = 2
BangkokData['shared_room']= int(0)

In [159]:
BangkokData['shared_room']=entire_shared_private_room(BangkokData)


In [160]:
for i in range(30):
    print(BangkokData.EntireHomeVsRoom[i],BangkokData['shared_room'][i])

ENTIRE APARTMENT 0
ENTIRE APARTMENT 0
ENTIRE CONDOMINIUM 0
ENTIRE CONDOMINIUM 0
ENTIRE APARTMENT 0
ENTIRE CONDOMINIUM 0
PRIVATE ROOM IN APARTMENT 2
ENTIRE APARTMENT 0
ENTIRE CONDOMINIUM 0
ENTIRE APARTMENT 0
ENTIRE APARTMENT 0
ENTIRE SERVICED APARTMENT 0
ENTIRE CONDOMINIUM 0
ROOM IN BOUTIQUE HOTEL 2
ENTIRE APARTMENT 0
ENTIRE APARTMENT 0
ENTIRE APARTMENT 0
ENTIRE CONDOMINIUM 0
ENTIRE APARTMENT 0
ENTIRE CONDOMINIUM 0
ENTIRE APARTMENT 0
ENTIRE CONDOMINIUM 0
ENTIRE APARTMENT 0
ENTIRE APARTMENT 0
ENTIRE CONDOMINIUM 0
ENTIRE APARTMENT 0
ENTIRE CONDOMINIUM 0
ENTIRE APARTMENT 0
ENTIRE LOFT 0
ENTIRE APARTMENT 0


### Bed Type

In [161]:
for i in BangkokData.Sleeping_engagment:
    if(str(i)!='nan'):
        if(i.split(' ')[2]=='small'):
            print(i)

In [162]:
BangkokData.Sleeping_engagment = BangkokData.Sleeping_engagment.str.replace('small double','double')

In [163]:
BangkokData['BedType']= None
BangkokData['BedType']= bed_type(BangkokData)

In [164]:
BangkokData['BedType'].unique()

array(['queen', 'double', 'king', None, 'sofa', 'floor', 'single',
       'couch', 'air', 'couch,', 'bunk'], dtype=object)

In [165]:
BangkokData = one_hot_encode_bedtype(BangkokData)

In [167]:
BangkokData.drop(['BedType','couch,'],axis=1, inplace=True)

### Amenities

In [168]:
BangkokData.Amenities.unique()

array(["['Wifi', 'Iron', 'Cable TV', 'Laptop friendly workspace', 'Dryer', 'Washer', 'Essentials', 'Air conditioning', 'TV', 'Hot water', 'Elevator', 'Gym', 'Pool', 'Kitchen', 'Coffee maker', 'Cooking basics', 'Dishes and silverware', 'Microwave', 'Refrigerator', 'Stove', 'Long term stays allowed', 'Hair dryer', 'Hangers', 'Shampoo', 'Smoke detector']",
       "['Wifi', 'Indoor fireplace', 'Heating', 'Iron', 'Laptop friendly workspace', 'Washer', 'Essentials', 'Air conditioning', 'TV', 'Elevator', 'Free parking on premises', 'Gym', 'Pool', 'Kitchen', 'Private entrance', 'Hair dryer', 'Hangers', 'Shampoo', 'Fire extinguisher', 'Carbon monoxide detector', 'Smoke detector', 'First aid kit']",
       "['Wifi', 'Heating', 'Laptop friendly workspace', 'Dryer', 'Washer', 'Essentials', 'Air conditioning', 'TV', 'Hot water', 'Elevator', 'Gym', 'Pool', 'Cooking basics', 'Dishes and silverware', 'Microwave', 'Refrigerator', 'Host greets you', 'Long term stays allowed', 'Hair dryer', 'Hangers', 'S

In [169]:
print(BangkokData.Amenities[0])

['Wifi', 'Iron', 'Cable TV', 'Laptop friendly workspace', 'Dryer', 'Washer', 'Essentials', 'Air conditioning', 'TV', 'Hot water', 'Elevator', 'Gym', 'Pool', 'Kitchen', 'Coffee maker', 'Cooking basics', 'Dishes and silverware', 'Microwave', 'Refrigerator', 'Stove', 'Long term stays allowed', 'Hair dryer', 'Hangers', 'Shampoo', 'Smoke detector']


In [170]:
BangkokData['Amenities_mod'] = None

In [171]:
BangkokData['Amenities_mod']= amenities_to_vars(BangkokData)

In [172]:
BangkokData['Amenities_mod'][0:3]

0    [Wifi, Iron, Cable TV, Laptop friendly workspa...
1    [Wifi, Indoor fireplace, Heating, Iron, Laptop...
2    [Wifi, Heating, Laptop friendly workspace, Dry...
Name: Amenities_mod, dtype: object

In [199]:
BangkokData = encode_amenities(BangkokData)

In [202]:
BangkokData.drop(['Amenities'],axis =1, inplace = True)

In [220]:
BangkokData.drop(['Title','EntireHomeVsRoom','Bedrooms_no','Beds_no','Bathrooms_num','Sleeping_engagment','Review'],axis=1 ,inplace = True)

### Cairo

### Review Stars

In [174]:
num_of_stars(CairoData)

CheckIn         CheckOut  \
0     2019-03-14 0:00  2019-03-19 0:00   
1     2019-03-14 0:00  2019-03-19 0:00   
2     2019-03-14 0:00  2019-03-19 0:00   
3     2019-03-14 0:00  2019-03-19 0:00   
4     2019-03-14 0:00  2019-03-19 0:00   
5     2019-03-14 0:00  2019-03-19 0:00   
6     2019-03-14 0:00  2019-03-19 0:00   
7     2019-03-14 0:00  2019-03-19 0:00   
8     2019-03-14 0:00  2019-03-19 0:00   
9     2019-03-14 0:00  2019-03-19 0:00   
10    2019-03-14 0:00  2019-03-19 0:00   
11    2019-03-14 0:00  2019-03-19 0:00   
12    2019-03-14 0:00  2019-03-19 0:00   
13    2019-03-14 0:00  2019-03-19 0:00   
14    2019-03-14 0:00  2019-03-19 0:00   
15    2019-03-14 0:00  2019-03-19 0:00   
16    2019-03-14 0:00  2019-03-19 0:00   
17    2019-03-14 0:00  2019-03-19 0:00   
18    2019-03-14 0:00  2019-03-19 0:00   
19    2019-03-14 0:00  2019-03-19 0:00   
20    2019-03-14 0:00  2019-03-19 0:00   
21    2019-03-14 0:00  2019-03-19 0:00   
22    2019-03-14 0:00  2019-03-19 0:00   
23    2019-03-14 0:00  2019-03-19 0:00   
24    2019-03-14 0:00  2019-03-19 0:00   
25    2019-03-14 0:00  2019-03-19 0:00   
26    2019-03-14 0:00  2019-03-19 0:00   
27    2019-03-14 0:00  2019-03-19 0:00   
28    2019-03-14 0:00  2019-03-19 0:00   
29    2019-03-14 0:00  2019-03-19 0:00   
...               ...              ...   
7872  2020-02-27 0:00  2020-03-03 0:00   
7873  2020-02-27 0:00  2020-03-03 0:00   
7874  2020-02-27 0:00  2020-03-03 0:00   
7875  2020-02-27 0:00  2020-03-03 0:00   
7876  2020-02-27 0:00  2020-03-03 0:00   
7877  2020-02-27 0:00  2020-03-03 0:00   
7878  2020-02-27 0:00  2020-03-03 0:00   
7879  2020-02-27 0:00  2020-03-03 0:00   
7880  2020-02-27 0:00  2020-03-03 0:00   
7881  2020-02-27 0:00  2020-03-03 0:00   
7882  2020-02-27 0:00  2020-03-03 0:00   
7883  2020-02-27 0:00  2020-03-03 0:00   
7884  2020-02-27 0:00  2020-03-03 0:00   
7885  2020-02-27 0:00  2020-03-03 0:00   
7886  2020-02-27 0:00  2020-03-03 0:00   
7887  2020-02-27 0:00  2020-03-03 0:00   
7888  2020-02-27 0:00  2020-03-03 0:00   
7889  2020-02-27 0:00  2020-03-03 0:00   
7890  2020-02-27 0:00  2020-03-03 0:00   
7891  2020-02-27 0:00  2020-03-03 0:00   
7892  2020-02-27 0:00  2020-03-03 0:00   
7893  2020-02-27 0:00  2020-03-03 0:00   
7894  2020-02-27 0:00  2020-03-03 0:00   
7895  2020-02-27 0:00  2020-03-03 0:00   
7896  2020-02-27 0:00  2020-03-03 0:00   
7897  2020-02-27 0:00  2020-03-03 0:00   
7898  2020-02-27 0:00  2020-03-03 0:00   
7899  2020-02-27 0:00  2020-03-03 0:00   
7900  2020-02-27 0:00  2020-03-03 0:00   
7901  2020-02-27 0:00  2020-03-03 0:00   

                                                  Title  Price  Super_host  \
0                                            Dad Haus!x   15.0           0   
1                       Apartment in Rehab City, Cairo.   12.0           0   
2     Sunny and friendly apartment close to the airport   22.0           0   
3     Cozy 2 persons budget room in luxurious/safe area   19.0           0   
4            Girls apartment - Sunny and calm apartment   23.0           0   
5                                      My little heaven   25.0           0   
6      PRIVATE TWIN OR DOUBLE ROOM. CITY VIEW.FREE WIFI   40.0           0   
7                       Cozy Apartment near the airport   37.0           1   
8     In prime location Studio in The Village,New Cairo   65.0           0   
9               Maadi club house ladies hostel room 304   11.0           1   
10           Comfortable & friendly room, close airport   14.0           0   
11                                           Star House   37.0           0   
12                        Cozy room 1 at heart of Cairo   13.0           1   
13        Star House Apartment, Nasr City, Cairo, Egypt   37.0           0   
14      Family Palace 2 beds room with private bathroom   25.0           0   
15             Expat apartment - Clean and cosy bedroom   31.0           0   
16            Spacious studio in the heart of New Cairo   45.0           1   
17   

In [176]:
for i in stars_list:
    print(CairoData[i].unique())


['0' '4' '5' '4.5' '3.5' '3']
['0' '5' '4.5' '4' '3.5']
['0' '3.5' '5' '4.5' '4' '3']
['0' '5' '4.5' '4' '3.5']
['0' '4.5' '5' '4' '3.5']
['0' '4' '5' '4.5' '3.5' '3']


### Entire Apartment vs. Shared Room vs. Private Room

In [177]:
CairoData.EntireHomeVsRoom.unique()

array(['Private room in apartment', 'Entire apartment',
       'Private room in condominium', 'Room in hotel',
       'Private room in hostel', 'Entire serviced apartment',
       'Entire condominium', 'Private room in bed and breakfast',
       'Entire house', 'Room in boutique hotel', 'Private room in house',
       'Private room in guesthouse', 'Private room in serviced apartment',
       'Shared room in hostel', 'Private room in tiny house',
       'Tiny house', 'Private room in villa', 'Shared room in apartment',
       'Entire villa', 'Earth house', 'Private room in guest suite',
       'Shared room in villa', 'Entire guest suite',
       'Private room in loft', 'Entire loft', 'Entire townhouse',
       'Private room in boat', 'Entire chalet', 'Shared room in house'],
      dtype=object)

In [178]:
CairoData['shared_room']= int(0)

In [179]:
CairoData['shared_room'] = entire_shared_private_room(CairoData)

In [180]:
for i in range(20):
    print(CairoData.EntireHomeVsRoom[i],CairoData['shared_room'][i])

Private room in apartment 2
Entire apartment 0
Private room in apartment 2
Private room in condominium 2
Private room in apartment 2
Private room in apartment 2
Room in hotel 2
Entire apartment 0
Entire apartment 0
Private room in hostel 2
Private room in apartment 2
Entire serviced apartment 0
Private room in apartment 2
Entire serviced apartment 0
Room in hotel 2
Private room in apartment 2
Entire condominium 0
Private room in bed and breakfast 2
Entire apartment 0
Private room in apartment 2


### Bed Type

In [181]:
for i in CairoData.Sleeping_engagment:
    if(str(i)!='nan'):
        if(i.split(' ')[2]=='small'):
            print(i)

Bedroom 1
2 small double beds
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed
Bedroom 1
1 small double bed


In [182]:
CairoData.Sleeping_engagment = CairoData.Sleeping_engagment.str.replace('small double','double')

In [183]:
CairoData['BedType']= None
CairoData['BedType']= bed_type(CairoData)

In [184]:
CairoData['BedType'].unique()

array(['queen', None, 'single', 'double', 'sofa', 'king', 'floor',
       'couch', 'couches', 'couch,', 'air', 'bunk'], dtype=object)

In [186]:
CairoData = one_hot_encode_bedtype(CairoData)

In [187]:
CairoData.drop(['couches', 'couch,','BedType'],axis=1, inplace=True)

### Amenities

In [188]:
CairoData.Amenities.unique()

array(["['Wifi', 'Indoor fireplace', 'Dryer', 'Laptop friendly workspace', 'Iron', 'Heating', 'Washer', 'TV', 'Air conditioning', 'Essentials', 'Elevator', 'Free parking on premises', 'Kitchen', 'Breakfast', 'Private living room', 'Hair dryer', 'Shampoo', 'Hangers', 'Lock on bedroom door', 'Fire extinguisher', 'Carbon monoxide detector', 'Smoke detector', 'First aid kit']",
       "['Wifi', 'Laptop friendly workspace', 'Washer', 'TV', 'Essentials', 'Hot water', 'Bathtub', 'Window guards', 'Elevator', 'Free parking on premises', 'Single level home', 'Free street parking', 'Kitchen', 'Microwave', 'Refrigerator', 'Oven', 'Stove', 'Host greets you', 'Luggage dropoff allowed', 'Long term stays allowed', 'Hangers', 'Extra pillows and blankets', 'Patio or balcony', 'Garden or backyard']",
       "['Wifi', 'Indoor fireplace', 'Cable TV', 'Laptop friendly workspace', 'Iron', 'Heating', 'Washer', 'TV', 'Air conditioning', 'Essentials', 'Ethernet connection', 'Hot water', 'Babysitter recommendati

In [189]:
CairoData.Amenities[0]

"['Wifi', 'Indoor fireplace', 'Dryer', 'Laptop friendly workspace', 'Iron', 'Heating', 'Washer', 'TV', 'Air conditioning', 'Essentials', 'Elevator', 'Free parking on premises', 'Kitchen', 'Breakfast', 'Private living room', 'Hair dryer', 'Shampoo', 'Hangers', 'Lock on bedroom door', 'Fire extinguisher', 'Carbon monoxide detector', 'Smoke detector', 'First aid kit']"

In [190]:
CairoData['Amenities_mod'] = None

In [191]:
CairoData['Amenities_mod'] = amenities_to_vars(CairoData)

In [192]:
CairoData.Amenities_mod[0]

['Wifi',
 'Indoor fireplace',
 'Dryer',
 'Laptop friendly workspace',
 'Iron',
 'Heating',
 'Washer',
 'TV',
 'Air conditioning',
 'Essentials',
 'Elevator',
 'Free parking on premises',
 'Kitchen',
 'Breakfast',
 'Private living room',
 'Hair dryer',
 'Shampoo',
 'Hangers',
 'Lock on bedroom door',
 'Fire extinguisher',
 'Carbon monoxide detector',
 'Smoke detector',
 'First aid kit']

In [203]:
CairoData= encode_amenities(CairoData)

In [205]:
CairoData.drop(['Amenities'],axis=1 ,inplace = True)

In [221]:
CairoData.drop(['Title','EntireHomeVsRoom','Bedrooms_no','Beds_no','Bathrooms_num','Sleeping_engagment','Review'],axis=1 ,inplace = True)

### Final Dataset

### NewYork

In [222]:
print(len(NewYorkData.columns))
NewYorkData.columns

107


Index(['Accuracy_stars', 'CheckIn', 'CheckIn_stars', 'CheckOut',
       'Cleanliness_stars', 'Communication_stars', 'Latitude',
       'Location_stars', 'Longitude', 'Price',
       ...
       'Smart lock', 'Smoke detector', 'Stair gates', 'Stove', 'TV',
       'Table corner guards', 'Washer', 'Waterfront', 'Wifi', 'Window guards'],
      dtype='object', length=107)

In [223]:
NewYorkData.head()

Accuracy_stars              CheckIn CheckIn_stars             CheckOut  \
0              5  2019-03-21 00:00:00             5  2019-03-26 00:00:00   
1            4.5  2019-03-21 00:00:00           4.5  2019-03-26 00:00:00   
2              5  2019-03-21 00:00:00             5  2019-03-26 00:00:00   
3              5  2019-03-21 00:00:00             5  2019-03-26 00:00:00   
4              5  2019-03-21 00:00:00             5  2019-03-26 00:00:00   

  Cleanliness_stars Communication_stars  Latitude Location_stars  Longitude  \
0                 5                   5  40.73321              5  -74.00242   
1                 4                 4.5  40.76580              5  -73.95812   
2                 5                   5  40.75972            4.5  -74.03827   
3                 5                   5  40.81386              5  -73.95936   
4                 5                   5  40.69284            4.5  -73.94418   

   Price      ...        Smart lock Smoke detector  Stair gates  Stove TV  \
0  244.0      ...                 0              1            0      1  1   
1   62.0      ...                 0              1            0      0  0   
2   44.0      ...                 0              1            0      0  0   
3   78.0      ...                 0              1            0      1  0   
4   80.0      ...                 0              1            0      0  0   

   Table corner guards  Washer  Waterfront  Wifi  Window guards  
0                    0       0           0     1              0  
1                    0       0           0     1              0  
2                    0       0           0     1              0  
3                    0       0           0     1              0  
4                    0       1           0     1              0  

[5 rows x 107 columns]

### Bangkok

In [224]:
print(len(BangkokData.columns))
BangkokData.columns

105


Index(['CheckIn', 'CheckOut', 'Price', 'Super_host', 'Latitude', 'Longitude',
       'Accuracy_stars', 'Communication_stars', 'Cleanliness_stars',
       'Location_stars',
       ...
       'Smart lock', 'Smoke detector', 'Stair gates', 'Stove', 'TV',
       'Table corner guards', 'Washer', 'Waterfront', 'Wifi', 'Window guards'],
      dtype='object', length=105)

In [225]:
BangkokData.head()

CheckIn         CheckOut  Price  Super_host  Latitude  Longitude  \
0  2019-03-12 0:00  2019-03-17 0:00   34.0           1  13.78165  100.57453   
1  2019-03-12 0:00  2019-03-17 0:00   48.0           0  13.69750  100.60708   
2  2019-03-12 0:00  2019-03-17 0:00   27.0           1  13.71252  100.46783   
3  2019-03-12 0:00  2019-03-17 0:00   35.0           0  13.72032  100.49623   
4  2019-03-12 0:00  2019-03-17 0:00   34.0           1  13.75738  100.56318   

  Accuracy_stars Communication_stars Cleanliness_stars Location_stars  \
0              5                   5               4.5              5   
1              0                   0                 0              0   
2              5                   5                 5              4   
3              5                   5               4.5              5   
4              0                   0                 0              0   

       ...       Smart lock Smoke detector  Stair gates  Stove TV  \
0      ...                0              1            0      1  1   
1      ...                0              1            0      0  1   
2      ...                0              1            0      0  1   
3      ...                0              1            0      1  1   
4      ...                0              0            0      0  1   

   Table corner guards  Washer  Waterfront  Wifi  Window guards  
0                    0       1           0     1              0  
1                    0       1           0     1              0  
2                    0       1           0     1              0  
3                    0       1           0     1              1  
4                    0       1           0     1              0  

[5 rows x 105 columns]

In [226]:
BangkokData.columns

Index(['CheckIn', 'CheckOut', 'Price', 'Super_host', 'Latitude', 'Longitude',
       'Accuracy_stars', 'Communication_stars', 'Cleanliness_stars',
       'Location_stars',
       ...
       'Smart lock', 'Smoke detector', 'Stair gates', 'Stove', 'TV',
       'Table corner guards', 'Washer', 'Waterfront', 'Wifi', 'Window guards'],
      dtype='object', length=105)

### Cairo

In [227]:
print(len(CairoData.columns))
CairoData.columns

106


Index(['CheckIn', 'CheckOut', 'Price', 'Super_host', 'Latitude', 'Longitude',
       'Accuracy_stars', 'Communication_stars', 'Cleanliness_stars',
       'Location_stars',
       ...
       'Smart lock', 'Smoke detector', 'Stair gates', 'Stove', 'TV',
       'Table corner guards', 'Washer', 'Waterfront', 'Wifi', 'Window guards'],
      dtype='object', length=106)

In [228]:
CairoData.head()

CheckIn         CheckOut  Price  Super_host  Latitude  Longitude  \
0  2019-03-14 0:00  2019-03-19 0:00   15.0           0  29.95961   31.25858   
1  2019-03-14 0:00  2019-03-19 0:00   12.0           0  30.05760   31.48779   
2  2019-03-14 0:00  2019-03-19 0:00   22.0           0  30.06085   31.49540   
3  2019-03-14 0:00  2019-03-19 0:00   19.0           0  30.06284   31.51743   
4  2019-03-14 0:00  2019-03-19 0:00   23.0           0  30.06263   31.48585   

  Accuracy_stars Communication_stars Cleanliness_stars Location_stars  \
0              0                   0                 0              0   
1              0                   0                 0              0   
2              4                   5               3.5            4.5   
3              5                   5                 5              5   
4              0                   0                 0              0   

       ...       Smart lock Smoke detector  Stair gates  Stove TV  \
0      ...                0              1            0      0  1   
1      ...                0              0            0      1  1   
2      ...                0              0            0      0  1   
3      ...                0              1            0      0  1   
4      ...                0              0            0      1  1   

   Table corner guards  Washer  Waterfront  Wifi  Window guards  
0                    0       1           0     1              0  
1                    0       1           0     1              1  
2                    0       1           0     1              0  
3                    0       1           0     1              0  
4                    0       1           0     1              0  

[5 rows x 106 columns]

## Saving Data

In [231]:
NewYorkData.to_csv(path_or_buf= Data_dir/'NewYorkData_clean.csv',columns=NewYorkData.columns,index=False)
BangkokData.to_csv(path_or_buf= Data_dir/'BangkokData_clean.csv',columns=BangkokData.columns,index=False)
CairoData.to_csv(path_or_buf= Data_dir/'CairoData_clean.csv',columns=CairoData.columns,index=False)